.github/workflows/formatting.yml

```diff
@@ -0,0 +1,24 @@
+name: Check Formatting
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+    formatting:
+      name: Check Formatting
+      runs-on: ubuntu-latest
+
+      steps:      
+        - name: Checkout code
+          uses: actions/checkout@v2
+        - name: Set system to non-interactive mode
+          run: export DEBIAN_FRONTEND=noninteractive
+        - name: install dependencies
+          run: |
+            sudo apt-get install -y --force-yes -qq git clang-format
```

Comment:
It would probably good at least to have the CI stick with a particular `clang-format`... could this be accomplished by changing this line to
```
sudo apt-get install -y --force-yes -qq git clang-format-10
```
?